src/Storages/StorageMergeTree.cpp

```diff
@@ -640,6 +640,12 @@ bool StorageMergeTree::merge(
 
         auto can_merge = [this, &lock] (const DataPartPtr & left, const DataPartPtr & right, String *) -> bool
         {
+            /// Ignore fingerprinted parts.
+            if ((left && !left->fingerprint.empty()) || !right->fingerprint.empty())
```

Comment:
Need the same logic for replicated merge tree.

Reply:
yep, the PR is based upon #14860 and should be covered there